diff -r -u a/ac.scm b/ac.scm
--- a/ac.scm	2011-02-06 09:57:34.471103777 -0800
+++ b/ac.scm	2011-02-06 09:59:06.385162230 -0800
@@ -646,11 +646,16 @@
 ; default vals with them.  To make compatible with existing written tables, 
 ; just use an atom or 3-elt list to keep the default.
 
+(defarc alref arc-alref)
+
 (define (ar-apply fn args)
   (cond ((procedure? fn) 
          (apply fn args))
         ((pair? fn) 
-         (list-ref fn (car args)))
+         ((if (number? (car args)) list-ref arc-alref)
+          fn (car args)))
+        ((eq? fn 'nil)
+         'nil)
         ((string? fn) 
          (string-ref fn (car args)))
         ((hash-table? fn) 
Only in b: ac.scm.orig
